sdk/storage/azure-storage-blobs/test/ut/bearer_token_test.cpp

```diff
@@ -22,4 +22,19 @@ namespace Azure { namespace Storage { namespace Test {
     EXPECT_NO_THROW(containerClient.Delete());
   }
 
-}}} // namespace Azure::Storage::Test
\ No newline at end of file
+  TEST_F(BlockBlobClientTest, BearerChallengeWorks_LIVEONLY_)
+  {
+    Blobs::BlobClientOptions clientOptions;
+    clientOptions.EnableTenantDiscovery = true;
+    Azure::Identity::ClientSecretCredentialOptions options;
+    options.AdditionallyAllowedTenants = {"*"};
+    auto blobClient = Blobs::BlobClient(
+        m_blockBlobClient->GetUrl(),
+        std::make_shared<Azure::Identity::ClientSecretCredential>(
+            AadTenantId(), AadClientId(), AadClientSecret(), options),
```

Comment:
How is this different from the test case above? You provided tenant ID. How does the new policy make things different?

Reply:
This tenantId is in credential,  and will not pass to StorageBearerTokenAuthenticationPolicy. So when enableTenantDiscovery, it will trigger bearer challenge.

Reply:
what if `enableTenantDiscovery` is not enable? `m_TenantId` is always initialized as empty. 

https://github.com/microzchang/azure-sdk-for-cpp/blob/6727a4381ae5e11e4e243a6f9911329dbb696405/sdk/storage/azure-storage-common/src/storage_bearer_token_authentication_policy.cpp#L22

is it a problem if this field at this line is empty?

Reply:
No, there is no problem when m_TenantId is empty. The tenantId is always empty in [the old logic](https://github.com/Azure/azure-sdk-for-cpp/blob/b33e8a9fd43a39762ac4e648336cd110c5fc58ab/sdk/storage/azure-storage-blobs/src/blob_client.cpp#L89C37-L89C37) and [ClientSecretCredential](https://github.com/Azure/azure-sdk-for-cpp/blob/b33e8a9fd43a39762ac4e648336cd110c5fc58ab/sdk/identity/azure-identity/src/client_secret_credential.cpp#L73) can [resolve tenantId from both credential and tokenRequestContext](https://github.com/Azure/azure-sdk-for-cpp/blob/b33e8a9fd43a39762ac4e648336cd110c5fc58ab/sdk/identity/azure-identity/src/tenant_id_resolver.cpp#L29).